lang.FormatCurrency & lang.FormatAccounting do not allow PRECISION < 2

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.87.0+extended darwin/amd64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?
yes.

Not sure if it is related to the "Add new localized versions of lang.FormatNumber etc." from v0.87.0, or if it is intended (not on doc)  but I can't get PRECISION = 0 or 1 for `lang.FormatCurrency` & `lang.FormatAccounting`

```
{{ 100 | lang.FormatCurrency 0 "CHF" }}<br>
{{ 100 | lang.FormatCurrency 1 "CHF" }}<br>
{{ 100 | lang.FormatCurrency 2 "CHF" }}<br>
{{ 100 | lang.FormatCurrency 3 "CHF" }}<br>
{{ 100 | lang.FormatAccounting 0 "CHF" }}<br>
{{ 100 | lang.FormatAccounting 1 "CHF" }}<br>
{{ 100 | lang.FormatAccounting 2 "CHF" }}<br>
{{ 100 | lang.FormatAccounting 3 "CHF" }}<br>
{{ 100 | lang.FormatNumber 0 }}<br>
{{ 100 | lang.FormatNumber 1 }}<br>
{{ 100 | lang.FormatNumber 2 }}<br>
{{ 100 | lang.FormatNumber 3 }}<br>
```

produce

```
100,00 CHF
100,00 CHF
100,00 CHF
100,000 CHF
100,00 CHF
100,00 CHF
100,00 CHF
100,000 CHF
100
100,0
100,00
100,000
```

I'm attempting to reproduce the issue via unit tests in my feature branch ([see the diff here](https://github.com/ptgott/hugo/compare/master..ptgott/issue-8858)). I've reproduced the issue with `*Namespace.FormatCurrency()` and `*Namespace.FormatAccounting()` when the precision is zero, though not one:

```
Running tool: /usr/local/bin/go test -timeout 30s -run ^TestFormatNumbers$ github.com/gohugoio/hugo/tpl/lang

--- FAIL: TestFormatNumbers (0.00s)
    --- FAIL: TestFormatNumbers/FormatAccounting (0.00s)
        /Users/paulgottschling/OneDrive/Programming/active-repos/hugo/tpl/lang/lang_test.go:118: 
            error:
              values are not equal
            got:
              "100,00 CHF"
            want:
              "100 CHF"
            stack:
              /Users/paulgottschling/OneDrive/Programming/active-repos/hugo/tpl/lang/lang_test.go:118
                c.Assert(gotFr, qt.Equals, "100 CHF")
            
    --- FAIL: TestFormatNumbers/FormatCurrency (0.00s)
        /Users/paulgottschling/OneDrive/Programming/active-repos/hugo/tpl/lang/lang_test.go:103: 
            error:
              values are not equal
            got:
              "100,00 CHF"
            want:
              "100 CHF"
            stack:
              /Users/paulgottschling/OneDrive/Programming/active-repos/hugo/tpl/lang/lang_test.go:103
                c.Assert(gotFr, qt.Equals, "100 CHF")
            
FAIL
FAIL	github.com/gohugoio/hugo/tpl/lang	0.024s
FAIL
```

I'm working on reproducing this via Hugo template (see my unit test in language_test.go), though I'm still getting acquainted with the repo and haven't managed to get the site to build in the test.

Reproduced this in a unit test that builds the site with the template provided in the issue description (https://github.com/ptgott/hugo/blob/97d12f274272b3fb6824a14cef2f8e0449b05569/hugolib/language_test.go#L141):

```
Running tool: /usr/local/bin/go test -timeout 30s -run ^TestLanguageCurrencyAccountingZeroAndOne$ github.com/gohugoio/hugo/hugolib

--- FAIL: TestLanguageCurrencyAccountingZeroAndOne (0.01s)
    /Users/paulgottschling/OneDrive/Programming/active-repos/hugo/hugolib/language_test.go:173: No match for "FormatCurrency with precision 0: 100 CHF<br>" in content for public/fr_ch/index.html
        
        
        FormatCurrency with precision 0: 100,00 CHF<br>
        FormatCurrency with precision 1: 100,00 CHF<br>
        FormatCurrency with precision 2: 100,00 CHF<br>
        FormatCurrency with precision 3: 100,000 CHF<br>
        FormatAccounting with precision 0: 100,00 CHF<br>
        FormatAccounting with precision 1: 100,00 CHF<br>
        FormatAccounting with precision 2: 100,00 CHF<br>
        FormatAccounting with precision 3: 100,000 CHF<br>	
        
        "\n\nFormatCurrency with precision 0: 100,00 CHF<br>\nFormatCurrency with precision 1: 100,00 CHF<br>\nFormatCurrency with precision 2: 100,00 CHF<br>\nFormatCurrency with precision 3: 100,000 CHF<br>\nFormatAccounting with precision 0: 100,00 CHF<br>\nFormatAccounting with precision 1: 100,00 CHF<br>\nFormatAccounting with precision 2: 100,00 CHF<br>\nFormatAccounting with precision 3: 100,000 CHF<br>\t\n"
FAIL
FAIL	github.com/gohugoio/hugo/hugolib	0.034s
FAIL
```

--- a/tpl/lang/lang.go
+++ b/tpl/lang/lang.go
@@ -80,8 +80,10 @@ func (ns *Namespace) FormatPercent(precision, number interface{}) (string, error
 	return ns.translator.FmtPercent(n, p), nil
 }
 
-// FormatCurrency returns the currency reprecentation of number for the given currency and precision
+// FormatCurrency returns the currency representation of number for the given currency and precision
 // for the current language.
+//
+// The return value is formatted with at least two decimal places.
 func (ns *Namespace) FormatCurrency(precision, currency, number interface{}) (string, error) {
 	p, n, err := ns.castPrecisionNumber(precision, number)
 	if err != nil {
@@ -94,8 +96,10 @@ func (ns *Namespace) FormatCurrency(precision, currency, number interface{}) (st
 	return ns.translator.FmtCurrency(n, p, c), nil
 }
 
-// FormatAccounting returns the currency reprecentation of number for the given currency and precision
+// FormatAccounting returns the currency representation of number for the given currency and precision
 // for the current language in accounting notation.
+//
+// The return value is formatted with at least two decimal places.
 func (ns *Namespace) FormatAccounting(precision, currency, number interface{}) (string, error) {
 	p, n, err := ns.castPrecisionNumber(precision, number)
 	if err != nil {

